### DISCUSSION TITLE
FR: pass some additional variables to alerting template

### DISCUSSION THREAD

**Discussion by edo1** [grafana/grafana#51812] [Alerting]
Tried to create a template in the new Grafana Alerting, but have some issues.

`ValueString` looks ugly in notifications ~and cannot be parsed with Go templates~.
Is it possible to add a non-string variable that could be iterated over in a template?

It would also be nice to have panel and dashboard names (only URLs are available for now)

**Update:** I managed to convert `ValueString` to something like what was in old alerts:
```
{{ reReplaceAll "[[][^]]*metric='{?([^}']*)}?'[^]]*value=([0-9]*([.][0-9]{1,3})?)[^]]*](, )?" "$1: $2\n" .ValueString }}
```
But I definitely do not like this way.

**Comment by edo1** [comment]
There are several complaints related to `ValueString` formatting, e. g.
https://community.grafana.com/t/how-to-clean-value-output-on-v9-like-previous-alerting-version-in-v8/68167
https://community.grafana.com/t/is-there-a-way-to-make-grafana-9-slack-alerts-prettier/67505
https://community.grafana.com/t/grafana-alerts-not-formatted-after-upgrade/68221
https://community.grafana.com/t/value-variable-in-alert-message/56721/5

**Comment by edo1** [comment]
Another thing that would be nice to have in the alert message template is the duration of the issue (the difference in seconds between `EndsAt` and `StartsAt`).
This can be used in conjunction with the already existing [humanizeDuration](https://grafana.com/docs/grafana/next/alerting/fundamentals/annotation-label/template-functions/#humanizeduration) function.

**Comment by bsper2** [comment]
Yes having the `ValueString` be returned as a KeyValue would be very nice.

**Comment by bsper2** [comment]
Having a `ValueString` that is easier to parse is especially helpful when trying to use the new multidimensional alerts. Since in that case a lot of the time you end up with a `reduce` expression and a `math` expression, and since `ValueString` returns "the labels and value of each reduced expression in the alert" that means the string ends up with lots of duplicate clutter.

For Example, this disk space alert:

![vivaldi_s0IqIkFHFn](https://user-images.githubusercontent.com/78050683/178376070-22a039d5-6ac4-4b09-91b8-45b7999ab3ec.png)


Ends up with a ValueString of:
```
Value: [ var='Disk Used' labels={host=headnode, path=/} value=71.84190430880307 ], [ var='__Alerting__' labels={host=headnode, path=/} value=1 ]
```

If `ValueString` was a KeyValue that would allow us to remove the `__Alerting__` expression which I don't need taking up space with a duplicate list of labels.

**Comment by natellium** [comment]
Hey there! We want to let you all know that we are moving to Github Issues for feature requests. You can read more about the move [here](https://github.com/grafana/grafana/discussions/73424).

I will be closing this discussion, but if you’re still interested in this request being implemented please don’t hesitate to open a [feature request](https://github.com/grafana/grafana/issues/new?assignees=&labels=&projects=&template=1-feature_requests.md). Sorry for the inconvenience, and thanks for your contributions!

cc @grafana/alerting-squad
